interview-questions.md

```diff
@@ -5,46 +5,65 @@ First, without external resources. Challenge yourself to answer from memory.
 
-  Researched answer:
 
 
 
-4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "GET"    route: /students          
+Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "POST"   route: /students       
+action: "GET" route: /students
 
-action: "GET"    route: /students/new
+The controller method for this route is "index". This method is looking to retrieve all of the instances in the table.
 
-action: "GET"    route: /students/2  
+action: "POST" route: /students
 
-action: "GET"    route: /students/2/edit    
+The controller method for this route is "create". This method adds a new instance of the model to the db table.
 
-action: "PATCH"  route: /students/2      
+action: "GET" route: /students/new
 
-action: "DELETE" route: /students/2      
+The controller method for this route is "new". New generates a form with expected inputs for each of the columns in the table.
 
+action: "GET" route: /students/2
 
+The controller method for this route is "show". Show retrieves a single instance of the table.
 
-5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+action: "GET" route: /students/2/edit
+
+The controller method for this route is "edit". Edit generates a form with expected inputs for each of the columns in the table.
+
+action: "PATCH" route: /students/2
+
+The controller method for this route is "update". Update renders any changes made to the data of a specific instance within the table.
+
+action: "DELETE" route: /students/2
+
+The controller method for this route is "destroy". Destroy deletes the specific instance in the table.
+
+As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about user stories.
+
+ Story: 1: As a user I can see all the cats in the database. 2: As a user I can update cats in the database. 3: As a user I can destroy a cat in the database. 4: As a user I can create a new animal in the database. 5: As a user I can create a sighting of a cat in the database. 6: As a user I can update a cat sighting in the database. 7: As a user I can destroy a cat sighting in the database. 8: As a user I can run a report to list all sightings during a given time period. 9: As a user I want to see a picture for each animal. 10: As a user I can update each cat bio.
```

Comment:
You definitely got the CRUD actions covered. Typically when writing user stories you would think about the user's experience. That the user could delete a cat rather than "destroy a cat in the database."